x-pack/solutions/search/plugins/search_query_rules/server/routes.ts

```diff
@@ -108,7 +110,73 @@ export function defineRoutes({ logger, router }: { logger: Logger; router: IRout
         headers: {
           'content-type': 'application/json',
         },
-        body: rulesetData,
+        body:
+          rulesetData ??
+          response.customError({
+            statusCode: 404,
+            body: 'Ruleset not found',
+          }),
+      });
+    })
+  );
+  router.put(
+    {
+      path: APIRoutes.QUERY_RULES_RULESET_ID,
+      options: {
+        access: 'internal',
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['manage_search_query_rules'],
+        },
+      },
+      validate: {
+        params: schema.object({
+          ruleset_id: schema.string(),
+        }),
+        query: schema.object({
+          forceWrite: schema.boolean({ defaultValue: false }),
+        }),
+      },
+    },
+    errorHandler(logger)(async (context, request, response) => {
+      const core = await context.core;
+      const {
+        client: { asCurrentUser },
+      } = core.elasticsearch;
+      const user = core.security.authc.getCurrentUser();
+      if (!user) {
+        return response.customError({
+          statusCode: 502,
+          body: 'Could not retrieve current user, security plugin is not ready',
+        });
+      }
+      const hasSearchQueryRulesPrivilege = await asCurrentUser.security.hasPrivileges({
+        cluster: ['manage_search_query_rules'],
+      });
+      if (!hasSearchQueryRulesPrivilege.has_all_requested) {
+        return response.forbidden({
+          body: "Your user doesn't have manage_search_query_rules privileges",
+        });
+      }
+      const rulesetId = request.params.ruleset_id;
+      const forceWrite = request.query.forceWrite;
+      const isExisting = await fetchQueryRulesRuleset(asCurrentUser, rulesetId);
```

Comment:
see the comment on `x-pack/solutions/search/plugins/search_query_rules/server/lib/fetch_query_rules_ruleset.ts`